templates_bw/sounds/display_pack.html

```diff
@@ -0,0 +1,31 @@
+{% load cache %}
+{% load display_sound %}
+
+{% if pack %}
+{% cache 3600 bw_display_pack pack.0.id %}
+{% with pack.0 as pack %}
```

Comment:
why is pack a list with 1 item here? Can't you make it an object or None?

Reply:
I had the same question and then I realized that this is the trick we use in `display_pack` to not evaluate the query until we're inside the HTML cache. Instead of doing `Pack.objects.get(id=123)` we do `Pack.objects.filter(id=123)` and the query is not evaluated until we do `{% with pack.0 as pack %}`. I thought it look like a hack but I could not think of an alternative...

Reply:
I'm not sure that this is a useful way to cache this item though. It seems to add a bunch of additional complexity, and results in multiple queries being made when the cache is empty (1 to get the list of ids, and then 1 to get each individual pack). Why can't you pass in a queryset which includes all packs, and then put a cache around the whole pack display block? (https://github.com/MTG/freesound/pull/1364/files/1c7884554345f8c9bb60e0a60a705061ca327ae7#diff-02f7f32f2c95d04f7189b1b0b2e771bcR100)

Reply:
I guess there are 2 things here. One is caching the HTML of an *individual* `display_pack`. If the same pack is shown in different pages, the HTML will be reused. The other is to cache the trending packs section of the front page in its entirety. 

Using the `pack.0` is the general strategy for caching `display_pack` contents, so this is a thing of the `display_pack` template tag.

I guess the idea here is that we'll cache the IDs of trending packs and then cache each pack HTML separately. Alternatively we could simply pass a queryset and then cache the whole section including the for loop of the queryset (that's what you suggest right?). This will be compatible with the `display_pack` templatetag as it also accepts `Pack` object as input. However the `pack.0` thing will still be used in other places. 

I guess the reason not to do pass queryset is because the computation of trending packs might take significant time depending on the rules we decide to apply and it's better to have it in the `create_front_page_caches` command (run by the cronjob). Then, because I guess you can't store queryset in the cache we use a list of ids. But maybe you can store queryset in the cache. I'm not sure if I ever tried.

What do you think would be best option here?

Reply:
We've had plenty of conversations about that here at the office. Will open a new PR with suggestions for improvement. For now we'll leave it as it is.